### PR TITLE
vector mask selection changes

### PR DESCRIPTION
**Pull request by pineapplespatula**
add enterFreeTransformPathMode command, and allow users to set the mode of vector tools to either path or shape. used by https://github.com/adobe-photoshop/spaces-design/pull/1970

### CODE REVIEW

File: src/lib/tool.js [hunk 1]
```diff
@@ -105,19 +105,21 @@ define(function (require, exports) {
...
+    var shapeVectorTool = 0,
+        pathVectorTool = 1;
     /**
-     * Resets the mode of shape tools back to "shape" from "path" or "pixel".
+     * sets the mode of shape tools  0=vtmShapeLayer, 1=vtmPath, 2=vtmFill
      * 
      * @return {PlayObject}
      */
-    var resetShapeTool = function () {
+    var setShapeTool = function (type) {
```
Comment (iwehrman) [inline src/lib/tool.js]: Can we call this `setShapeToolMode` instead?

File: src/lib/tool.js [hunk 2]
```diff
@@ -105,19 +105,21 @@ define(function (require, exports) {
     };
 
 
+    var shapeVectorTool = 0,
```
Comment (iwehrman) [inline src/lib/tool.js]: These should be collected as part of an exported object and documented as such. Something like:
```javascript
toolModes = {
  SHAPE: 0,
  PATH: 1
}
```
And then in your -design branch do something like `toolLib.setShapeToolMode(toolLib.toolModes.PATH)`.

File: src/lib/tool.js [hunk 3]
```diff
@@ -105,19 +105,21 @@ define(function (require, exports) {
...
+        pathVectorTool = 1;
     /**
-     * Resets the mode of shape tools back to "shape" from "path" or "pixel".
+     * sets the mode of shape tools  0=vtmShapeLayer, 1=vtmPath, 2=vtmFill
      *
```
Comment (iwehrman) [inline src/lib/tool.js]: Also, this now needs an `@param` declaration.

File: src/lib/vectorMask.js [hunk 4]
```diff
@@ -137,9 +137,34 @@ define(function (require, exports) {
...
     };
     
+    /**
+     * free transform the whole path of the targeted vector mask
+     * 
+     * @return {PlayObject}
+     */
+    var enterFreeTransformPathMode = function () {
+        var layerRef = {
+                    "_ref": "layer",
```
Comment (iwehrman) [inline src/lib/vectorMask.js]: Tabbing issues in these two variable initializations.

File: src/lib/tool.js (hunk 1 above)
Comment (iwehrman) [inline src/lib/tool.js]: `setShapeToolMode`

**Comment by pineapplespatula** [comment]
@iwehrman  back to you

File: src/lib/vectorMask.js [hunk 5]
```diff
@@ -137,9 +137,62 @@ define(function (require, exports) {
...
     };
     
+    /**
+     * free transform the whole path of the targeted vector mask
+     * 
+     * @return {PlayObject}
+     */
+    var enterFreeTransformPathMode = function () {
+        var layerRef = {
+            "_ref": "layer",
```
Comment (iwehrman) [inline src/lib/vectorMask.js]: From here to the end of the declaration block should be indented one tab stop.

File: src/lib/tool.js [hunk 6]
```diff
@@ -105,19 +105,26 @@ define(function (require, exports) {
     };
 
 
+    var toolModes = {
```
Comment (iwehrman) [inline src/lib/tool.js]: JSDoc needed on this guy.

**Comment by pineapplespatula** [comment]
@iwehrman once more.

File: src/lib/tool.js [hunk 7]
```diff
@@ -104,20 +104,29 @@ define(function (require, exports) {
         );
     };
 
+    /**
+     * Tool Mode possible values
+     */
```
Comment (iwehrman) [inline src/lib/tool.js]: Need a @type annotation here.

Comment (iwehrman) [inline src/lib/tool.js]: `@type {Object.<string, number>}`, specifically.

File: src/lib/vectorMask.js [hunk 8]
```diff
@@ -137,9 +137,62 @@ define(function (require, exports) {
...
+                _ref: "property",
+                _property: "freeTransformWholePath"
+            };
+        
+        return new PlayObject("set", {
+            "null": {
+                "_ref": [propertyRef, layerRef]
+            },
+            "_property": "freeTransformWholePath",
+            "suppressPlayLevelIncrease": true
```
Comment (iwehrman) [inline src/lib/vectorMask.js]: I thought this was supposed to be option supplied to the play command? As opposed to this being just another property on the descriptor? Did I misunderstand?

Comment (pineapplespatula) [inline src/lib/vectorMask.js]: I'll double check with Jesper, but i think this is correct

**Comment by pineapplespatula** [comment]
@iwehrman back to you

**Comment by iwehrman** [comment]
:+1: :+1: :+1:

**Pull request merged**